### PR TITLE
Query page basic ui

### LINKED ISSUE #3: Query page UI [1~2 day]
Query page should display the predetermined advertiser id and the insertion order id. This will be changed later to display input box to allow for user interaction.

The query page should also include a submit-query button.

Cost estimate: 1~2 day

### LINKED ISSUE #8: Update query page UI with user input advertiser id [ < 1 day]
Update the query page to allow users input an advertiser id that is within their credential access right.

Cost estimate: < 1 day

### LINKED ISSUE #12: Update query page UI with user input advertiser id & media plan id [1 day]
In addition to advertiser id, also allow users to enter a media plan id.

Cost estimate: 1 day

### PR DESCRIPTION
**Pull request by hanj7221**
Closes #3.

Query page UI:
Screen Shot 2020-07-23 at 3 31 00 PM

Display error when user enters non-number into the id fields:
Screen Shot 2020-07-23 at 3 31 20 PM

Display error when user doesn't enter the required parameters:
Screen Shot 2020-07-23 at 3 46 36 PM

Populate button has a spinner that indicates request is still in progress:
Screen Shot 2020-07-23 at 3 31 51 PM

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -16,6 +17,7 @@ dependencies:
   fixnum: ^0.10.11
   js: ^0.6.1+1
   intl: ^0.15.8
+  ng_bootstrap: any
```
Comment (hanj7221) [inline pubspec.yaml:20]: Found a material design package that works, but had to downgrade angular and angular forms

**Review by arulprakash-g — APPROVED** [review]

File: lib/root_component.html [hunk 2]
```diff
@@ -1,3 +1,9 @@
-<h2 debugId="title">{{title}}</h2>
+<nav class="title-row shadow-sm navbar navbar-light bg-white">
+    <a class="navbar-brand" href="#" id="title" debugId="title">
+        <img src="https://developers.google.com/ads/images/logo_display_video_360_192px.svg"
```
Comment (ruben-aghayan) [inline lib/root_component.html]: can we provide the image as a local resource?

File: lib/src/query_component.dart [hunk 3]
```diff
@@ -18,31 +19,51 @@ import 'util.dart';
...
+  static bool showAlert = false;
+  static bool showSpinner = false;
+
   final QueryService _queryService;
   final ExcelDart _excel;
 
   // Default selection is byAdvertiser.
-  QueryType queryType = QueryType.byAdvertiser;
-  List<QueryType> queryTypeChoices = QueryType.values;
+  static QueryType queryType = QueryType.byAdvertiser;
```
Comment (ruben-aghayan) [inline lib/src/query_component.dart]: actually think this is more appropriate as an instance type. this is the currently selected query type right?

File: lib/src/query_component.dart [hunk 4]
```diff
@@ -83,6 +107,25 @@ class QueryComponent implements OnInit {
   /// Used in [query_component.html] as part of the radio button id.
   String getQueryTypeShortName(QueryType queryType) => queryType.shortName;
 
+  bool _missingRequiredParameters() {
+    switch (queryType) {
+      case QueryType.byAdvertiser:
```
Comment (ruben-aghayan) [inline lib/src/query_component.dart]: there is shared logic here, can we make use of fall through?

File: lib/src/query_component.dart [hunk 5]
```diff
@@ -83,6 +107,25 @@ class QueryComponent implements OnInit {
...
+        return advertiserId == null || advertiserId.isEmpty;
+
+      case QueryType.byMediaPlan:
```
Comment (ruben-aghayan) [inline lib/src/query_component.dart]: can i click even when i have alphabet text entered? the alert will show but is there anything stopping from the request being made?

File: lib/src/query_component.html [hunk 6]
```diff
@@ -1,17 +1,81 @@
...
+        <bs-accordion [closeOthers]="true">
+            <bs-accordion-panel class='query-type-name'
+                                id="{{getQueryTypeShortName(choice)}}-panel"
+                                *ngFor="let choice of queryTypeChoices"
+                                heading="{{getQueryTypeName(choice)}}"
+                                [isOpen]="queryType===choice"
+                                (click)="queryType=choice">
+                <div class="section-name">{{requestSectionTitle}}</div>
+                <form class="form-group">
+                    <div>
```
Comment (ruben-aghayan) [inline lib/src/query_component.html]: why doesnt this need an ngif like the other two

File: web/main.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+.underpacing-checkbox-name {
+  font: 16px Roboto, "Helvetica Neue", sans-serif;
+}
+
+.btn.btn-success {
+  text-transform: uppercase;
+  font: 13px Roboto, sans-serif;;
+}
+
+.btn.btn-primary{
```
Comment (ruben-aghayan) [inline web/main.scss:59]: try to use one font family for everything

File: web/main.scss [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+.parameter-input {
+  color: $regular-font-color;
+  font: 14px "Roboto Mono", monospace;
+}
+
+.underpacing-checkbox {
+  margin-top: 10px;
+  margin-bottom: 10px;;
+}
+.underpacing-checkbox-name {
```
Comment (ruben-aghayan) [inline web/main.scss]: you can put this inside .underpacing-checkbox

**Review by ruben-aghayan — CHANGES_REQUESTED** [review]
any tests? there is some non-css logic like the spinner and input validation. also showing the inputs fields mutually exclusively.

File: lib/src/query_component.dart (hunk 5 above)
Comment (hanj7221) [reply lib/src/query_component.dart]: Ohh good catch, maybe I should disable the button until all inputs are filled in & valid. Seems like the error message should be used when I get to error handling for failed query request.

File: lib/src/query_component.html (hunk 6 above)
Comment (hanj7221) [reply lib/src/query_component.html]: Since advertiser id is required no matter what query type it is.

File: web/main.scss [hunk 9]
```diff
@@ -1,55 +1,62 @@
 $title-section-background-color: #202124;
 $body-background-color: #E9ECEF;
 
+$title-font: "Google Sans";
```
Comment (hanj7221) [inline web/main.scss:4]: So three font families: one for the title, one for the body, and one for the parameter input box.

File: web/main.scss (hunk 7 above)
Comment (hanj7221) [reply web/main.scss:59]: see [here](https://github.com/googleinterns/dv360-excel-plugin/pull/66#discussion_r460973727)

**Review by hanj7221 — COMMENTED** [review]
How should I test for the spinner? It is removed(ngif) by default, and only when the query still in the processing stage it is added back.

File: lib/src/query_component.dart [hunk 10]
```diff
@@ -22,47 +22,69 @@ import 'util.dart';
...
+  /// if the ids are not integers.
+  ///
+  /// Binds to the [disabled] attribute on the populate button.
+  bool disablePopulateButton() {
+    var invalid = false;
+    final numberOnly = RegExp(numberOnlyPattern);
+    switch (queryType) {
+      case QueryType.byMediaPlan:
+        invalid = mediaPlanId == null || !numberOnly.hasMatch(mediaPlanId);
+        continue checkAdvertiser;
```
Comment (ruben-aghayan) [inline lib/src/query_component.dart:68]: woah thats cool

does io not need media plan id?

File: test/query_component_test.dart [hunk 11]
```diff
@@ -60,9 +61,9 @@ class QueryTestComponent {}
 void main() {
   group('In $QueryComponent,', () {
     const emptyEntry = '';
-    const advertiserId = 'advertiser-id';
-    const mediaPlanIdA = 'media-plan-id-a';
-    const ioIdA = 'io-id-a';
+    const advertiserId = '10000';
+    const mediaPlanIdA = '20000';
```
Comment (ruben-aghayan) [inline test/query_component_test.dart]: opt: we typically use number e.g. mediaPlanId1

File: test/query_component_test.dart [hunk 12]
```diff
@@ -182,33 +183,37 @@ void main() {
...
     });
 
-    group('selecting ByAdvertiser with', () {
-      const mediaPlanIdB = 'media-plan-id-b';
-      const ioIdB = 'io-id-b';
-      const ioIdC = 'io-id-c';
-      const ioIdD = 'io-id-d';
+    group('selecting ByAdvertiser panel with', () {
+      const mediaPlanIdB = '20001';
+      const ioIdB = '30001';
```
Comment (ruben-aghayan) [inline test/query_component_test.dart]: instantiate these up with the other test consts

File: test/query_component_test.dart [hunk 13]
```diff
@@ -249,18 +254,32 @@ void main() {
         ];
       });
 
-      test(
-          'highlight underpacing unchecked invokes populate(multiple-IO, false)',
+      tearDown(() => clearInteractions(mockQueryService));
```
Comment (ruben-aghayan) [inline test/query_component_test.dart:303]: this shouldnt be necessary if you are making a new instance for every test

File: test/query_component_test.dart [hunk 14]
```diff
@@ -11,6 +11,7 @@ import 'package:dv360_excel_plugin/src/query_service.dart';
 import 'package:dv360_excel_plugin/src/util.dart';
```
Comment (ruben-aghayan) [inline test/query_component_test.dart:11]: should test the spinner somewhere

File: lib/src/query_component.dart (hunk 10 above)
Comment (hanj7221) [reply lib/src/query_component.dart:68]: haha nope :)

File: test/query_component_test.dart (hunk 11 above)
Comment (hanj7221) [reply test/query_component_test.dart:65]: Since i check for the input format, having non-integer ids will actually disable the button

**Review by ruben-aghayan — APPROVED** [review]

File: test/query_component_test.dart (hunk 13 above)
Comment (hanj7221) [reply test/query_component_test.dart:303]: I'm using singleton pattern (needs component provider overwrite) here so it always returns the same instance.

**Pull request merged**